readers.go

```diff
@@ -17,8 +19,10 @@ func readIntoTree(t *Tree, r io.Reader, segmentSize int) error {
 			break
 		} else if readErr == io.ErrUnexpectedEOF {
 			// This is the last segment, and there aren't enough bytes to fill
-			// the entire segment. Note that the next call will return io.EOF.
+			// the entire segment. We break here because io.Reader does not
+			// enforce that io.EOF is returned after io.ErrUnexpectedEOF.
 			segment = segment[:n]
+			break
```

Comment:
go doc io.Reader says
> [after an error such as io.ErrUnexpectedEOF] The next Read should return 0, EOF.

But this is not enforced so a buggy reader might continue returning io.ErrUnexpectedEOF. Is it reasonable to break here for extra safety?